src/Monkeypatcher.cc

```diff
@@ -817,7 +817,14 @@ void Monkeypatcher::patch_after_mmap(RecordTask* t, remote_ptr<void> start,
     ScopedFd open_fd = t->open_fd(child_fd, O_RDONLY);
     ASSERT(t, open_fd.is_open()) << "Failed to open child fd " << child_fd;
     ElfFileReader reader(open_fd, t->arch());
-    auto syms = reader.read_symbols(".symtab", ".strtab");
+    ScopedFd debug_fd = reader.open_debug_file(map.map.fsname());
+    SymbolTable syms;
+    if (debug_fd.is_open()) {
+      ElfFileReader debug_reader(debug_fd, t->arch());
+      syms = debug_reader.read_symbols(".symtab", ".strtab");
+    } else {
+      syms = reader.read_symbols(".symtab", ".strtab");
+    }
```

Comment:
I think it's probably better/safer to check both files, the main file first. I don't think we can assume the debuglink file will have `.symtab`.